activerecord/lib/active_record/core.rb

```diff
@@ -98,6 +98,13 @@ def self.error_on_ignored_order_or_limit=(value)
         self.error_on_ignored_order = value
       end
 
+      # :singleton-method:
+      # Specify the behavior for unsafe raw query methods. A value of true will
+      # rename methods like #order to #unsafe_raw_order and will validate
+      # arguments to #order. A value of false will do nothing.
+      mattr_accessor :guard_unsafe_raw_sql, instance_writer: false
+      self.guard_unsafe_raw_sql = false
```

Comment:
Agree it should be opt-out by using the `unsafe_*` API.

However, for compatibility with existing code, would expect to be able to turn off checks on a model superclass: `SomeEngine::Model.guard_unsafe_raw_sql = false` without affecting app models or other engines.